### ISSUE TITLE
Question: Plans for Deterctron2?

### ISSUE THREAD

**Issue by abfleishman** [obss/sahi#176]
I would love to use sliced inference for a `detectron2` model (Mask-RCNN).  I tried to build a `mmdetection` model but failed to get past some issues that kept popping up.  I am new to `detectron2` and I am not sure where to start to attempt to make `sahi` work with the models I am training.  Are there plans to extend `sahi` to work with detectron2?  if not, do you have advice for me on where to start to attempt to extend it myself?

**Comment by fcakyon** [comment]
Actually mmdetection is more stable than detectron2 and covers much more state-of-the-art models.

Are you on windows or linux?
What is your python version?
What issues did you have with mmdetection?

**Comment by fcakyon** [comment]
you can check the contributions sections of the readme for extending the support: https://github.com/obss/sahi#contributing

**Comment by abfleishman** [comment]
Thank you! 

I have been using Linux Ubuntu VMs on Azure.  I am using a `conda` environment for `mmdetection` and I am running `python 3.7.10`.  I installed following their tutorial and things appear to have installed correctly (though I have not successfully trained a model so maybe not).

I have had trouble tracking down the issues as I am new to the `mmdetection` package, new to the `coco` data format, and new to models that predict masks.  However, I think I have overcome the data-prep when using normal 3channel imagery (slicing tiffs, prepping coco labels) issues because I was able to train a model (`Mask-RCNN` with only one class) using `Detectron2` and it works pretty well. I have not gotten the 4 channel images to work yet.

The issue I am having with `mmdetection` likely stems from the model not working well and not producing any proposals.  The model trains for 1 epoch and then errors on the evaluation. I get the same error that is reported here: https://github.com/open-mmlab/mmdetection/issues/1067#issuecomment-529414103

I have attempted to fool with the hyper parameters a bit but it didn't seem to help much and could not get past the error.

**Comment by fcakyon** [comment]
@abfleishman sometimes people arrange config files for `mmdetection` incorrectly (mostly the dataset configs), are you sure all is fine on that part? can you show the config files you prepared for `mmdetection` training? most of the times mmdetection runs train much stable and faster than detectron2. this is why in obss we utilize mmdetection models more than detectron2 models in production.

**Comment by abfleishman** [comment]
@fcakyon Thank you for your willingness to look this over for me!  I have found the learning curve steeper than expected for `mmdetection` Please see the attached. config that I am using (I changed the extension from `py` to `txt` to have github upload it)
[eseal_config.txt](https://github.com/obss/sahi/files/6896936/eseal_config.txt)
  This second file is the  output from running the `trainer.train()` 
[config.txt](https://github.com/obss/sahi/files/6896923/config.txt)

**Comment by fcakyon** [comment]
@abfleishman all seems to be great. only advice i could give is, if you are training with single gpu then you should increase the learning rate:
from `optimizer = dict(type='SGD', lr=0.002, momentum=0.9, weight_decay=0.0001)` to `optimizer = dict(type='SGD', lr=0.016, momentum=0.9, weight_decay=0.0001)`

**Comment by abfleishman** [comment]
@fcakyon Thank you!  I made the change and it is training now! It is amazing how such a small tweak can fix an error.  Now that I have the `mmdetection` model training with RGB tiles, now the real work begins. I have two tasks ahead of me: 1) get the model to train with 4 channel images, 2) use the `sahi` package to run the model on the large ortho mosaics 10,000x10,000 px.

**Comment by fcakyon** [comment]
@abfleishman i have a plan to work on large tiff image processing in the following days but 4 channel model is all up to you. You can combine some of the channels to convert your 4channel images into 3 channels or you can directly disregard the less important channel and continue with the remaining 3 channels.

**Comment by abfleishman** [comment]
yes thank you for your help!  I would have given up on `mmdetection` with out you!

**Comment by fcakyon** [comment]
@abfleishman i totally understand you, `mmdetection` has a high entry barrier but once you get used to it, there is not turning back 😄

**Comment by fcakyon** [comment]
closing this since this is not a priority for us, feel free to open a pr

**Comment by fcakyon** [comment]
there is a pr for this now: https://github.com/obss/sahi/pull/322

**Comment by FinnMers** [comment]
Thank you! Do you already know when you will merge the PR?

**Comment by fcakyon** [comment]
@FinnMers we are planning to merge it next week :rocket:
